File: KartMayhem.Model/Kupovina.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KartMayhem.Model
{
    public class Kupovina
    {
        public int Id { get; set; }
        public int Cijena { get; set; }
        public DateTime? DatumKupovine { get; set; }
        public string? PaymentIntentId { get; set; }
        public bool? Placena { get; set; }
        public int? KorisnikId { get; set; }
        public virtual Korisnici? Korisnik { get; set; }
        public int? StazaId { get; set; }
        public virtual Staze? Staze { get; set; }
    }
}

File: UI/kartmayhem_mobile/lib/Models/kupovina.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:kartmayhem_mobile/Models/korisnik.dart';
import 'package:kartmayhem_mobile/Models/staze.dart';

part 'kupovina.g.dart';

@JsonSerializable()
class Kupovina {
  int? id;
  int? kolicina;
  int? cijena;
  DateTime datumKupovine;
  String? paymentIntentId;
  int? korisnikId;
  Korisnik? korisnik;
  int? stazeId;
  Staze? staze;
  bool? placena;

  Kupovina(
      {required this.id,
      this.kolicina,
      this.cijena,
      required this.datumKupovine,
      this.paymentIntentId,
      this.korisnikId,
      this.korisnik,
      this.stazeId,
      this.staze,
      this.placena});

  factory Kupovina.fromJson(Map<String, dynamic> json) =>
      _$KupovinaFromJson(json);
  Map<String, dynamic> toJson() => _$KupovinaToJson(this);
}
